### PR TITLE
Resolves StackVisualizer is Obsolete

### PR DESCRIPTION
**Pull request by TemporalOroboros**
## About the PR

Splits the StackVisualizer into a component/system pair.

### TODO:
- [ ] Testing

**Media**

- [ ] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

### CODE REVIEW

File: Content.Client/Stack/StackVisualizerComponent.cs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+///       - cigarette_4
+///       - cigarette_5
+///       - cigarette_6
+/// </code>
+/// </example>
+///  <seealso cref="_spriteLayers"/>
+/// </summary>
+[RegisterComponent]
+[Access(typeof(StackVisualizerSystem))]
+public sealed class StackVisualizerComponent : Component
```
Comment (AJCM-git) [inline Content.Client/Stack/StackVisualizerComponent.cs:43]: move to stack component (also VV fields and format attributes above the fields, not in the same line)

File: Content.Client/Stack/StackVisualizerSystem.cs [hunk 2]
```diff
@@ -0,0 +1,89 @@
+using Content.Shared.Rounding;
+using Content.Shared.Stacks;
+using Robust.Client.GameObjects;
+using Robust.Shared.Utility;
+
+namespace Content.Client.Stack;
+
+public sealed class StackVisualizerSystem : VisualizerSystem<StackVisualizerComponent>
```
Comment (AJCM-git) [inline Content.Client/Stack/StackVisualizerSystem.cs:8]: move to stacksystem

File: Content.Client/Stack/StackVisualizerSystem.cs [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+{
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<StackVisualizerComponent, ComponentInit>(OnInit);
```
Comment (AJCM-git) [inline Content.Client/Stack/StackVisualizerSystem.cs:13]: oncomponentinit

File: Content.Client/Stack/StackVisualizerSystem.cs [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+            ProcessOpaqueSprites(uid, comp, args.Component, args.Sprite);
+        }
+    }
+
+    private void ProcessOpaqueSprites(EntityUid uid, StackVisualizerComponent comp, AppearanceComponent appearance, SpriteComponent spriteComponent)
+    {
+        // Skip processing if no actual
+        if(!AppearanceSystem.TryGetData<int>(uid, StackVisuals.Actual, out var actual, appearance))
+            return;
+        if(!AppearanceSystem.TryGetData<int>(uid, StackVisuals.MaxCount, out var maxCount, appearance))
```
Comment (AJCM-git) [inline Content.Client/Stack/StackVisualizerSystem.cs:52]: newline

File: Content.Client/Stack/StackVisualizerSystem.cs [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+                spriteComponent.LayerSetVisible(transparentSprite, false);
+            }
+
+            return;
+        }
+
+        // Skip processing if no actual/maxCount
+        if(!AppearanceSystem.TryGetData<int>(uid, StackVisuals.Actual, out var actual, appearance))
+            return;
+        if(!AppearanceSystem.TryGetData<int>(uid, StackVisuals.MaxCount, out var maxCount, appearance))
```
Comment (AJCM-git) [inline Content.Client/Stack/StackVisualizerSystem.cs:77]: newline

**Review by AJCM-git — DISMISSED** [review]

**Comment by DrSmugleaf** [comment]
Reopen when you come back to this.

File: Content.Client/Stack/StackVisualizerComponent.cs (hunk 1 above)
Comment (TemporalOroboros) [reply Content.Client/Stack/StackVisualizerComponent.cs:43]: The logic is also used by `ItemCounterComponent`. Do you want me to copypasta to its system as well?

**Comment by TemporalOroboros** [comment]
@DrSmugleaf 
Commits tested and pushed. Please reopen whenever's convenient.

**Pull request merged**